### PR TITLE
Split into multiple plugins

### PR DESCRIPTION
**Pull request by loosebazooka**
- App Engine Plugin (Decider, Applyable)
- App Engine Core Plugin (has deploy and defines the "appengine" extension, and "deploy")
- Standard Plugin (has stage and run for standard apps)
- Flexible Plugin (has stage for flex apps)
~- Source Context Plugin (Applyable)~ moved to another CL

This change also changes how extensions are created. Using nested
extensions instead of evaluation closures.

### CODE REVIEW

**Comment by loosebazooka** [comment]
@meltsufin @patflynn 

I really really apologize for this, but when writing the SourceContext Plugin, there wasn't a good way to really apply it nicely. I decided to split the plugins out.

Most of the new code is in "com.google.cloud.tools.gradle.appengine.sourcecontext" and everything else is just moving a bunch of stuff around.

The mechanism for instantiating extensions changed a little. Testing seems to think it's okay. In the end, the public API hasn't changed, so people don't need to change their build files.

File: src/main/java/com/google/cloud/tools/gradle/appengine/base/extension/Deploy.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
  *
  */
 
-package com.google.cloud.tools.gradle.appengine.model;
+package com.google.cloud.tools.gradle.appengine.base.extension;
```
Comment (patflynn) [inline src/main/java/com/google/cloud/tools/gradle/appengine/base/extension/Deploy.java]: I'm not a big fan of 'base' as a package name.  Maybe 'common' or if it's for a provided framework maybe 'framework'?

File: src/main/java/com/google/cloud/tools/gradle/appengine/base/task/ShowConfigurationTask.java [hunk 2]
```diff
@@ -15,42 +15,35 @@
...
-  public Class<?> getExtensionClass() {
-    return extensionClass;
-  }
-
-  public void setExtensionClass(Class<?> extension) {
-    this.extensionClass = extension;
-  }
-
-  @Input
   public Object getExtensionInstance() {
```
Comment (patflynn) [inline src/main/java/com/google/cloud/tools/gradle/appengine/base/task/ShowConfigurationTask.java]: this shit blows my mind, but I don't think I have time to try and understand

File: src/main/java/com/google/cloud/tools/gradle/appengine/base/extension/Deploy.java (hunk 1 above)
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/gradle/appengine/base/extension/Deploy.java]: It was a reflection of how gradle has some of their plugins (JavaBasePlugin), anyway, I changed it to core now.

**Comment by loosebazooka** [comment]
Also if this review is annoying or painful, I can pull out the sourcecontext stuff and add it to a follow up PR.

File: src/main/java/com/google/cloud/tools/gradle/appengine/base/task/ShowConfigurationTask.java [hunk 3]
```diff
@@ -15,42 +15,35 @@
...
+import java.util.Map;
 
 /**
  * Task to print the appengine configuration closure
  */
 public class ShowConfigurationTask extends DefaultTask {
 
-  // require extensionClass because gradle creates a decorated_class extension instance
-  private Class<?> extensionClass;
+  private static String MODEL_PKG = "com.google.cloud.tools.gradle.appengine";
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/gradle/appengine/base/task/ShowConfigurationTask.java]: word

**Comment by loosebazooka** [comment]
moved source context out of this PR into another one.

**Pull request merged**